7_react/react1/week2/todo-list-app/src/TodoList.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import TodoHeader from './TodoHeader';
+import TodoItem from './TodoItem';
+
+const todos = [
+  {
+    id: 1,
+    description: "Get out of bed",
+    done: false,
+  },
+  {
+    id: 2,
+    description: "Brush teeth",
+    done: false,
+  },
+  {
+    id: 3,
+    description: "Eat breakfast",
+    done: false,
+  },
+];
+
+const TodoList = () => {
+  const [todoList, setList] = useState(todos);
+
+  const toggleDone = (id) => {
+    setList((prevList) =>
+      prevList.map((item) =>
+        item.id === id ? { ...item, done: !item.done } : item
```

Comment:
wow nice! If you don't fully understand these lines, feel free to reach out to me :) I know this can be tricky.